nntrainer/layers/layer_context.h

```diff
@@ -142,7 +142,8 @@ class InitLayerContext {
                              const Tensor::Initializer init,
                              const WeightRegularizer reg, const float reg_const,
                              const std::string &name, bool trainable = true) {
-    weights_spec.emplace_back(dim, init, reg, reg_const, trainable, name);
+    weights_spec.emplace_back(dim, init, reg, reg_const, trainable,
+                              getName() + ":" + name);
```

Comment:
In this way, we need to guarantee those tensors are only used inside of layer then. Just for sure, Is there no case that call or expose beyond the layer such as network??

Reply:
The tensors can be used externally as well. This patch changes the name which is given to the tensor at creation time. So if any code caches the name or uses it, they will use the correct name.